src/gui_common/CustomDropDown.cs

```diff
@@ -0,0 +1,165 @@
+using System.Collections.Generic;
+using Godot;
+
+/// <summary>
+///   Basically a class that inherits MenuButton, but with extra popup menu functionality
+///   such as adjusted custom icon size with tweakable color and some slide down animation.
+///   (Might need to expand this later)
+/// </summary>
+public class CustomDropDown : MenuButton
+{
+    /// <summary>
+    ///   The MenuButton's popup menu
+    /// </summary>
+    public PopupMenu Popup;
+
+    private Tween tween;
+
+    /// <summary>
+    ///   All item icon sizes will be adjusted according to this. Currently it's automatically
+    ///   set according to the PopupMenu's check icon size (with a bit smaller result)
+    /// </summary>
+    private Vector2 iconSize;
+
+    private float cachedPopupHSeparation;
+    private float cachedPopupVSeparation;
+
+    private List<Item> items;
+
+    public CustomDropDown()
+    {
+        Popup = GetPopup();
+        items = new List<Item>();
+        tween = new Tween();
+
+        AddChild(tween);
+
+        cachedPopupHSeparation = Popup.GetConstant("hseparation");
+        cachedPopupVSeparation = Popup.GetConstant("vseparation");
+
+        var checkSize = Popup.GetIcon("checked").GetSize();
+
+        // Set the custom icon size
+        iconSize = new Vector2(checkSize.x - 2, checkSize.y - 2);
+
+        Popup.RectClipContent = true;
+
+        Connect("about_to_show", this, nameof(OnPopupAboutToShow));
+        Popup.Connect("draw", this, nameof(RedrawPopup));
+    }
+
+    /// <summary>
+    ///   Helper for adding item into PopupMenu and also an icon to be custom drawn in this class
+    /// </summary>
+    public void AddItem(string text, int id, bool checkable, Color color, Texture icon = null)
+    {
+        var item = new Item
+        {
+            Text = text,
+            Icon = icon,
+            Color = color,
+            Id = id,
+            Checkable = checkable,
+        };
+
+        items.Add(item);
+
+        if (checkable)
+        {
+            Popup.AddCheckItem(item.Text, item.Id);
+        }
+        else
+        {
+            Popup.AddItem(item.Text, id);
+        }
+
+        // Waits until next frame, this fixes an incorrect rect size for some reason
+        Invoke.Instance.Queue(() => Popup.EmitSignal("draw"));
```

Comment:
I think it's supposed to work so that you mark something as dirty and it will automatically redraw on next frame. At least I think I read that in the godot docs.